physics/module_mp_thompson.F90

```diff
@@ -92,142 +92,142 @@ MODULE module_mp_thompson
 !.. problems with Paul Field's moments and should not be set larger
 !.. because of truncation problems in snow/ice growth.
-      REAL, PARAMETER, PRIVATE:: R1 = 1.E-12
-      REAL, PARAMETER, PRIVATE:: R2 = 1.E-6
-      REAL, PARAMETER :: eps = 1.E-15
+      REAL(kind_phys), PARAMETER, PRIVATE:: R1 = 1.E-12
+      REAL(kind_phys), PARAMETER, PRIVATE:: R2 = 1.E-6
+      REAL(kind_phys), PARAMETER :: eps = 1.E-15
 
 !..Constants in Cooper curve relation for cloud ice number.
-      REAL, PARAMETER, PRIVATE:: TNO = 5.0
-      REAL, PARAMETER, PRIVATE:: ATO = 0.304
+      REAL(kind_phys), PARAMETER, PRIVATE:: TNO = 5.0
+      REAL(kind_phys), PARAMETER, PRIVATE:: ATO = 0.304
 
 !..Rho_not used in fallspeed relations (rho_not/rho)**.5 adjustment.
-      REAL, PARAMETER, PRIVATE:: rho_not = 101325.0/(287.05*298.0)
+      REAL(kind_phys), PARAMETER, PRIVATE:: rho_not = 101325.0/(287.05*298.0)
 
 !..Schmidt number
-      REAL, PARAMETER, PRIVATE:: Sc = 0.632
-      REAL, PRIVATE:: Sc3
+      REAL(kind_phys), PARAMETER, PRIVATE:: Sc = 0.632
+      REAL(kind_phys), PRIVATE:: Sc3
 
 !..Homogeneous freezing temperature
-      REAL, PARAMETER, PRIVATE:: HGFR = 235.16
+      REAL(kind_phys), PARAMETER, PRIVATE:: HGFR = 235.16
 
 !..Water vapor and air gas constants at constant pressure
-      REAL, PARAMETER, PRIVATE:: Rv = 461.5
-      REAL, PARAMETER, PRIVATE:: oRv = 1./Rv
-      REAL, PARAMETER, PRIVATE:: R = 287.04
-      REAL, PARAMETER, PRIVATE:: Cp = 1004.0
-      REAL, PARAMETER, PRIVATE:: R_uni = 8.314                           !< J (mol K)-1
+      REAL(kind_phys), PARAMETER, PRIVATE:: Rv = 461.5
+      REAL(kind_phys), PARAMETER, PRIVATE:: oRv = 1./Rv
+      REAL(kind_phys), PARAMETER, PRIVATE:: R = 287.04
+      REAL(kind_phys), PARAMETER, PRIVATE:: Cp = 1004.0
+      REAL(kind_phys), PARAMETER, PRIVATE:: R_uni = 8.314                           !< J (mol K)-1
 
       DOUBLE PRECISION, PARAMETER, PRIVATE:: k_b = 1.38065E-23           !< Boltzmann constant [J/K]
       DOUBLE PRECISION, PARAMETER, PRIVATE:: M_w = 18.01528E-3           !< molecular mass of water [kg/mol]
       DOUBLE PRECISION, PARAMETER, PRIVATE:: M_a = 28.96E-3              !< molecular mass of air [kg/mol]
       DOUBLE PRECISION, PARAMETER, PRIVATE:: N_avo = 6.022E23            !< Avogadro number [1/mol]
       DOUBLE PRECISION, PARAMETER, PRIVATE:: ma_w = M_w / N_avo          !< mass of water molecule [kg]
-      REAL, PARAMETER, PRIVATE:: ar_volume = 4./3.*PI*(2.5e-6)**3        !< assume radius of 0.025 micrometer, 2.5e-6 cm
+      REAL(kind_phys), PARAMETER, PRIVATE:: ar_volume = 4./3.*PI*(2.5e-6)**3        !< assume radius of 0.025 micrometer, 2.5e-6 cm
 
 !..Enthalpy of sublimation, vaporization, and fusion at 0C.
-      REAL, PARAMETER, PRIVATE:: lsub = 2.834E6
-      REAL, PARAMETER, PRIVATE:: lvap0 = 2.5E6
-      REAL, PARAMETER, PRIVATE:: lfus = lsub - lvap0
-      REAL, PARAMETER, PRIVATE:: olfus = 1./lfus
+      REAL(kind_phys), PARAMETER, PRIVATE:: lsub = 2.834E6
```

Comment:
@gthompsnWRF Regarding your note in the discussion about performance, would you say that everything in your scheme that is not explicitly defined as double-precision should be single-precision? (If so then I would suggest @dongli replace the changes here to use the kind_sngl_phys and kind_dbl_phys types, rather than kind_phys everywhere)

Reply:
We need to use https://github.com/NCAR/ccpp-physics/pull/918 as a guide. `kind_phys` needs to be kept for variables where single precision is acceptable, but can `kind_phys` can be set to either single or double precision depending on the user's needs. The issue is that someone actually needs to figure out which variables MUST stay in double precision rather than just adding `kind_phys` to everything, since that doesn't necessarily help anything (except perhaps with other hosts' compilation).